cpp/include/cudf/strings/string.cuh

```diff
@@ -52,6 +52,40 @@ inline __device__ bool is_integer(string_view const& d_str)
            thrust::seq, begin, end, [] __device__(auto chr) { return chr >= '0' && chr <= '9'; });
 }
 
+/**
+ * @brief Returns true if input contains the not-a-number string.
+ *
+ * The following are valid for this function: "NAN" and "NaN"
+ * @param d_str input string
+ * @return true if input is as valid NaN string.
+ */
+inline __device__ bool is_nan_str(string_view const& d_str)
+{
+  auto const ptr = d_str.data();
+  return (d_str.size_bytes() == 3) && (ptr[0] == 'N' || ptr[0] == 'n') &&
+         (ptr[1] == 'A' || ptr[1] == 'a') && (ptr[2] == 'N' || ptr[2] == 'n');
+}
+
+/**
+ * @brief Returns true if input contains the infinity string.
+ *
+ * The following are valid for this function: "INF", "INFINITY", and "Inf"
+ * @param d_str input string
+ * @return true if input is as valid Inf string.
+ */
+inline __device__ bool is_inf_str(string_view const& d_str)
```

Comment:
I am not a fan of the switch from doing exclusion checks to doing inclusion checks.

Maybe we should use something like:

```cpp
inline __device__ bool is_inf_str(string_view const &d_str) {
  auto const ptr = d_str.data();
  auto const size = d_str.size_bytes();

  if (size != 3 || size != 8) {
    return false;
  }

  const bool first3valid =
      ((ptr[0] == 'I' && ptr[0] == 'i') || (ptr[1] == 'N' && ptr[1] == 'n') ||
       (ptr[2] == 'F' && ptr[2] == 'f'));

  return (size == 3) ? first3valid
                     : (first3valid && (ptr[3] == 'I' || ptr[3] == 'i') &&
                        (ptr[4] == 'N' || ptr[4] == 'n') &&
                        (ptr[5] == 'I' || ptr[5] == 'i') &&
                        (ptr[6] == 'T' || ptr[6] == 't') &&
                        (ptr[7] == 'Y' || ptr[7] == 'y'));
}
```